src/lib/addons/datadog.ts

```diff
@@ -9,6 +9,12 @@ import {
 } from './feature-event-formatter-md';
 import { IEvent } from '../types/events';
 
+interface IDatadogParameters {
+    url: string;
+    apiKey: string;
+    customHeaders?: string;
```

Comment:
Why not something like:
```
    customHeaders?: { [key: string]: string } | Record<string, string> ;
```
That would save you from doing `JSON.parse` later

Inspired by https://github.com/search?q=%22type+HeadersInit+%3D%22+language%3ATypeScript&type=code&l=TypeScript

Reply:
Hmm, if it works that would be good, we just don't have a way to define the input parameter in the addon definition as anything but a string. So somewhere we'd need to do the parsing.